Makefile

```diff
@@ -229,39 +234,101 @@ build-all-in-one-debug build-agent-debug build-query-debug build-collector-debug
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/ingester/ingester$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/ingester/
 
 .PHONY: build-remote-storage build-remote-storage-debug
 build-remote-storage build-remote-storage-debug:
-	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/remote-storage/remote-storage$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/remote-storage/main.go
+	$(GOBUILD) $(DISABLE_OPTIMIZATIONS) -o ./cmd/remote-storage/remote-storage$(SUFFIX)-$(GOOS)-$(GOARCH) $(BUILD_INFO) ./cmd/remote-storage/
 
 .PHONY: build-binaries-linux
 build-binaries-linux:
 	GOOS=linux GOARCH=amd64 $(MAKE) build-platform-binaries
 
+# Magic values:
+# - LangID "0409" is "US-English".
+# - CharsetID "04B0" translates to decimal 1200 for "Unicode".
+# - FileOS "040004" defines the Windows kernel "Windows NT".
+# - FileType "01" is "Application".
+define VERSIONINFO
+{
+    "FixedFileInfo": {
+        "FileVersion": {
+            "Major": $(GIT_CLOSEST_TAG_MAJOR),
+            "Minor": $(GIT_CLOSEST_TAG_MINOR),
+            "Patch": $(GIT_CLOSEST_TAG_PATCH),
+            "Build": 0
+        },
+        "ProductVersion": {
+            "Major": $(GIT_CLOSEST_TAG_MAJOR),
+            "Minor": $(GIT_CLOSEST_TAG_MINOR),
+            "Patch": $(GIT_CLOSEST_TAG_PATCH),
+            "Build": 0
+        },
+        "FileFlagsMask": "3f",
+        "FileFlags ": "00",
+        "FileOS": "040004",
+        "FileType": "01",
+        "FileSubType": "00"
+    },
+    "StringFileInfo": {
+        "FileDescription": "NAME",
+        "FileVersion": "$(GIT_CLOSEST_TAG_MAJOR).$(GIT_CLOSEST_TAG_MINOR).$(GIT_CLOSEST_TAG_PATCH).0",
+        "LegalCopyright": "2015-2023 The Jaeger Project Authors",
+		"ProductName": "NAME",
+        "ProductVersion": "$(GIT_CLOSEST_TAG_MAJOR).$(GIT_CLOSEST_TAG_MINOR).$(GIT_CLOSEST_TAG_PATCH).0"
+    },
+    "VarFileInfo": {
+        "Translation": {
+            "LangID": "0409",
+            "CharsetID": "04B0"
+        }
+    }
+}
+endef
+
+export VERSIONINFO
+
+.PHONY: prepare-winres
+prepare-winres:
+	NAME="Jaeger Agent"				PKGPATH="cmd/agent"				$(MAKE) prepare-winres-helper
+	NAME="Jaeger Collector" 		PKGPATH="cmd/collector" 		$(MAKE) prepare-winres-helper
+	NAME="Jaeger Query"				PKGPATH="cmd/query"				$(MAKE) prepare-winres-helper
+	NAME="Jaeger Ingester"			PKGPATH="cmd/ingester"			$(MAKE) prepare-winres-helper
+	NAME="Jaeger Remote Storage"	PKGPATH="cmd/remote-storage"	$(MAKE) prepare-winres-helper
+	NAME="Jaeger All-In-One"		PKGPATH="cmd/all-in-one"		$(MAKE) prepare-winres-helper
+	NAME="Jaeger Tracegen"			PKGPATH="cmd/tracegen"			$(MAKE) prepare-winres-helper
+	NAME="Jaeger Anonymizer"		PKGPATH="cmd/anonymizer"		$(MAKE) prepare-winres-helper
+	NAME="Jaeger ES-Index-Cleaner"	PKGPATH="cmd/es-index-cleaner"	$(MAKE) prepare-winres-helper
+	NAME="Jaeger ES-Rollover"		PKGPATH="cmd/es-rollover"		$(MAKE) prepare-winres-helper
```

Comment:
The formatting looks messed up, can you use spaces instead of tabs (except the first one which has to be a tab)
<img width="1574" alt="image" src="https://github.com/jaegertracing/jaeger/assets/3523016/7152bf04-4bdc-4975-93e0-60aa04eec407">